changelog/deprecateNonPropertyAssignment.dd

```diff
@@ -0,0 +1,16 @@
+the property assignment to non-@property functions has been deprecated.
+
+Since it can result it can result in fairly unintutive errors
```

Comment:
Replace with: `Rationale: in an assignment, it is ambiguous whether the property should be invoked with no argument and the result be assigned to, or the property should be invoked with one argument.`